Replicator/Worker.hh

```diff
@@ -123,6 +130,8 @@ namespace litecore { namespace repl {
         virtual void afterEvent() override;
         virtual void caughtException(const std::exception &x) override;
 
+        fleece::Retained<C4Collection> collection();
```

Comment:
Perhaps add a comment that this method will fire assert if called on Replicator, whose collectionIndex, from Worker's perspective, is kNotCollectionIndex

Reply:
It will only do so in debug mode so I want to avoid that specific wording, see updated commit though.